OTAnalytics/plugin_ui/customtkinter_gui/toplevel_progress.py

```diff
@@ -1,34 +1,51 @@
-from typing import Any
+from typing import Any, Sequence
 
 from customtkinter import CTkLabel, CTkProgressBar, CTkToplevel
 
+from OTAnalytics.adapter_ui.abstract_progressbar_popup import (
+    AbstractPopupProgressbar,
+    ProgressbarPopupBuilder,
+)
+from OTAnalytics.application.progress import AutoIncrementingProgressbar, SimpleCounter
+from OTAnalytics.domain.progress import (
+    Counter,
+    ProgressbarBuilder,
+    ProgressbarBuildError,
+)
 from OTAnalytics.plugin_ui.customtkinter_gui.constants import PADX, PADY
+from OTAnalytics.plugin_ui.customtkinter_gui.helpers import get_widget_position
 
 
 class InvalidRelativeProgressError(Exception):
     pass
 
 
-class ToplevelProgress(CTkToplevel):
+class ProgressbarPopupTemplate(AbstractPopupProgressbar, CTkToplevel):
     def __init__(
         self,
+        counter: Counter,
+        unit: str,
+        total: int,
         initial_position: tuple[int, int],
-        initial_message: str,
-        initial_progress: float = 0.0,
+        description: str,
         title: str = "",
         **kwargs: Any,
     ) -> None:
+        self._unit = unit
+        self._total = total
+        self._current_progress = counter
+        self._close = False
+
         super().__init__(**kwargs)
         self.title(title)
-        self._initial_message = initial_message
-        self._initial_progress = initial_progress
+        self._description = description
         self._get_widgets()
         self._place_widgets()
         self._set_focus()
         self.protocol("WM_DELETE_WINDOW", self._on_cancel)
         self.overrideredirect(True)  # TODO: Test on Windows
```

Comment:
The MinimalInfoBox uses a direct call to tkinter instead of the wrapper. Is this also an issue here?

Reply:
There had been no complications so far with this widget, but definitely better to call it directly as well.